_protected/app/system/modules/mail/controllers/MainController.php

```diff
@@ -37,6 +37,9 @@ class MainController extends Controller
 
     /** @var int */
     private $iProfileId;
+    
+    /** @var int */
+    private $iSender;
```

Comment:
FYI, this attribute can be removed, since it isn't used anywhere in the class.

Reply:
Yes I forgot to remove it, I started off with an idea of how I was going to get the senders info but changed my mind and never removed it. 

I think removing the trash all together is a good idea. Simply mark as deleted until both users have deleted at which point it can be dropped form the database. 

Have you viewed it via a browser? You will understand why I removed the echo where, by and sort if you do. I was unable to pass the sender and subject to the mailform even though it is displayed on the same page and couldn't pass values in the use of the mailform() function either. I need to think on this a little more.